File: 2021/day6/question1.py
from input import *

input = input.split(",")
input = [int(i, base=10) for i in input]


for j in range(80):
    for i in range(len(input)):
        if input[i] == 0:
            input.append(8)
            input[i] = 6
        else:
            input[i] -= 1


print(len(input))

File: 2021/day3/question1.jl
include("input.jl")

input = split(input, "\n")

gamma = ""
epsilon = ""

for bit in 1:length(input[1])
    num0 = 0
    num1 = 0
    for byte in input
        if byte[bit] == '1'
            num1+=1
        else
            num0+=1
        end
    end
    global gamma *= string(Int(num0>num1))
    global epsilon *= string(Int(num0<num1))

end

println(parse(Int,gamma,base=2)*parse(Int,epsilon,base=2))
